alchemy/src/stripe/tax-rate.ts

```
@@ -0,0 +1,198 @@
+   */
+  created: number;
+  /**
+   * Has the value true if the object exists in live mode or the value false if the object exists in test mode
+   */
+  livemode: boolean;
+}
+
+/**
+ * Create and manage Stripe tax rates
+ *
+ * @example
+ * // Create a basic sales tax rate
+ * const salesTax = await TaxRate("ca-sales-tax", {
+ *   displayName: "CA Sales Tax",
+ *   percentage: 8.5,
+ *   inclusive: false,
+ *   active: true,
+ *   country: "US",
+ *   state: "CA",
+ *   description: "California sales tax",
+ *   taxType: "sales_tax"
+ * });
+ *
+ * @example
+ * // Create a VAT tax rate
+ * const vatTax = await TaxRate("uk-vat", {
+ *   displayName: "UK VAT",
+ *   percentage: 20.0,
+ *   inclusive: true,
+ *   active: true,
+ *   country: "GB",
+ *   description: "United Kingdom Value Added Tax",
+ *   taxType: "vat",
+ *   metadata: {
+ *     region: "europe",
+ *     tax_authority: "hmrc"
+ *   }
+ * });
+ *
+ * @example
+ * // Create a city tax rate
+ * const cityTax = await TaxRate("nyc-tax", {
+ *   displayName: "NYC Local Tax",
+ *   percentage: 4.5,
+ *   inclusive: false,
+ *   active: true,
+ *   country: "US",
+ *   state: "NY",
+ *   jurisdiction: "New York City",
+ *   description: "New York City local sales tax",
+ *   taxType: "sales_tax",
+ *   metadata: {
+ *     locality: "nyc",
+ *     tax_level: "city"
+ *   }
+ * });
+ */
+export const TaxRate = Resource(
+  "stripe::TaxRate",
+  async function (
+    this: Context<TaxRate>,
+    _id: string,
+    props: TaxRateProps,
+  ): Promise<TaxRate> {
+    const apiKey = process.env.STRIPE_API_KEY;
+    if (!apiKey) {
+      throw new Error("STRIPE_API_KEY environment variable is required");
+    }
+
+    const stripe = new Stripe(apiKey);
+
+    if (this.phase === "delete") {
+      try {
+        if (this.output?.id) {
+          await stripe.taxRates.update(this.output.id, { active: false });
+        }
+      } catch (error) {
+        console.error("Error deactivating tax rate:", error);
+      }
```

do not swallow errors. that is bad. only error that can be swallowed during a deletion is Not Found. go look up stripe api docs and make sure you detect that properly